### DISCUSSION TITLE
Typescript+React alternative to UseFormRegister<any> for generic FieldValues

### DISCUSSION THREAD

**Discussion by anders-photowall** [react-hook-form/react-hook-form#6190] [Q&A] [answered]
Hello, I'm trying to make a react component accept a register function for any FieldValues. I.e. in the parent component I want to be able to create my register function with `{ register, ... } = useForm<SpecifcFieldValuesInterfaceForMyCurrentForm>` and pass that to a more generic child that accept any register function that has a "FieldValues-like" interface. 

I'm a typescript novice so It might be some obvious typescript syntax or concept I'm missing, but I don't understand why setting the type of the generic components prop to`UseFormRegister<FieldValues>` doesn't work as my _SpecifcFieldValuesInterfaceForMyCurrentForm_ is of that type, although a bit more specific in that it has some specific records specified. 

Setting prop type to  `UseFormRegister<any>` does of course work but it's something I'd like to avoid.

**Comment by jorisre** [comment]
Hi! Can you share a codesandbox with your issue, please?

**Reply by anders-photowall** [reply]
Of course, here it is: https://codesandbox.io/s/cool-voice-gn99e?file=/src/App.tsx

**Comment by jorisre** [comment, marked as answer]
I recommend forwarding the ref of the input component. It's more consistent and easier to type.

Here is an instance: https://codesandbox.io/s/laughing-bas-eccqj

**Reply by anders-photowall** [reply]
Thank you, this approach was much better than what we used previously. After refactoring quite a bit it now works and has a more logical structure where the main benefit is that our "extended input" components, or whatever you want to call, them no longer knows anything about react-hook-forms but instead forwards their refs to the component that is actually using useForm(). 👍

**Reply by codinginflow** [reply]
Can't you just pass `UseFormRegisterReturn` instead of `UseFormRegister`? Then you call register on the outside and get the type-safety.
The code is also much simpler than `forwardRef`.
